pkg/bisect/bisect_test.go

```diff
@@ -107,16 +170,39 @@ func TestBisectCause(t *testing.T) {
 			KernelSrc:    repo.Dir,
 		},
 		Kernel: KernelConfig{
-			Commit: head.Hash,
-			Repo:   originRepo.Dir,
+			Repo: originRepo.Dir,
 		},
 	}
 	inst := &testEnv{
-		repo: repo,
-		r:    r,
-		t:    t,
+		repo:         repo,
+		r:            r,
+		t:            t,
+		crashAll:     crashAll,
+		fix:          fix,
+		inconclusive: inconclusive,
+	}
+	c := &Ctx{
+		t:       t,
+		baseDir: baseDir,
+		repo:    repo,
+		r:       r,
+		cfg:     cfg,
+		inst:    inst,
+	}
+	return c
+}
+
+// TestBisectCause tests that bisection returns the correct cause
+// commit.
+func TestBisectCause(t *testing.T) {
+	c := NewCtx(t, false, false, false)
```

Comment:
It's very common to use table-driven tests in Go b/c they allow to avoid lots of repetition and make code easier to change in future. See for example:
https://github.com/google/syzkaller/blob/d01bb02a96019cda0fa8c46e5c6d5eb66a273f17/prog/hints_test.go#L37-L170
I think it will be a very good fit for this case as well b/c there are lots of repetition in all tests, adding a new test requires a bunch of copy-pasting, extending bisect.Run return values will require lots of changes across all tests.
We need a test structure that will contain crashAll, fit, inconclusive and start commit and expected results. And then a single test that will be completely data driven.